lib/commands/exam.js

```diff
@@ -124,35 +135,90 @@ module.exports = TestCommand.extend({
    * @override
    */
   _generateCustomConfigs: function(commandOptions) {
-    var config = this._super._generateCustomConfigs.apply(this, arguments);
-
-    if (this.validator.shouldParallelize) {
-      // Get the testPage from the generated config or the Testem config and
-      // use it as the baseUrl for the parallelized test pages
-      var baseUrl = config.testPage || this._getTestPage(commandOptions.configFile);
-      var count = commandOptions.split;
-      var partitions = commandOptions.partition;
-
-      if (!partitions) {
-        partitions = [];
-        for (var i = 0; i < commandOptions.split; i++) {
-          partitions.push(i + 1);
-        }
-      }
+    const config = this._super._generateCustomConfigs.apply(this, arguments);
+    const customBaseUrl = this._getCustomBaseUrl(config, commandOptions);
+    const shouldParallelize = this.validator.shouldParallelize;
+    const shouldLoadBalance = this.validator.shouldLoadBalance;
 
-      if (Array.isArray(baseUrl)) {
-        var command = this;
-        config.testPage = baseUrl.reduce(function(testPages, baseUrl) {
-          return testPages.concat(command._generateTestPages(baseUrl, partitions, count));
-        }, []);
-      } else {
-        config.testPage = this._generateTestPages(baseUrl, partitions, count);
+    config.additionalBrowserSocketEvents = this._addBrowserSocketEventsToConfig(config);
```

Comment:
shouldn't this only be done when `shouldLoadBalance` is true?

Reply:
that's correct. Moved ` config.additionalBrowserSocketEvents = this._addBrowserSocketEventsToConfig(config);` under if (shouldLoadBalance) statement.